lib/client.js

```diff
@@ -137,6 +137,11 @@ Request.prototype.send = function() {
       }
     });
   } else {
+    // label limited to 63 chars and 253 chars max in total length
+    if(this.question.name.length > 253 || !this.question.name.match(/^(?:(?:-|\w|\d\.){1,63}\.)+(?:-|\w|\d\.){1,63}$/)){
+      this.emit('error',new Error(consts.BADNAME));
```

Comment:
@rogerc Could you add a comment explaining:

1. Where these restrictions comes from.
2. What the regex is doing, along with maybe some examples of the names it's designed to match (and not match)?

Sorry, it's just not super clear to me and it would help me to merge this PR faster.